api/src/classes/pagination.py

```diff
@@ -36,13 +41,35 @@ class CompanyPagination(Pagination):
 
     def __init__(
         self,
-        page: int = PaginationDefaults.PAGE.value,
-        limit: int = PaginationDefaults.LIMIT.value,
+        page: int = Check(default=PaginationDefaults.PAGE.value, ge=1),
+        limit: int = Check(default=PaginationDefaults.LIMIT.value, ge=1),
+        *,
+        ascending: bool = False,
+    ) -> None:
+        super().__init__(page=page, limit=limit)
+        self.ascending: bool | None = ascending
+
+    def as_dict(self) -> dict[str, Any]:
+        """Get the params as a dict."""
+        return super().as_dict()
+
+
+class ShopPagination(Pagination):
+    """Pagination for getting Shop Items."""
+
+    def __init__(
+        self,
+        page: int = Check(default=PaginationDefaults.PAGE.value, ge=1),
+        limit: int = Check(default=PaginationDefaults.LIMIT.value, ge=1),
         *,
         ascending: bool = False,
+        sort_by: list[str] | None = Check(default=None, examples=["price", "quantity"]),
```

Comment:
sort_by should have a default value, and shouldn't require a list
```diff
- sort_by: list[str] | None = Check(default=None, examples=["price", "quantity"]),
+ sort_by: str = Check(default="price", examples=["price", "quantity"]),
```

Reply:
If I want to sort by multiple options? Items with the cheapest price and the most quantity?

Reply:
That wouldn't be possible tho

Reply:
How wouldn't it be?

I have the following items:
Price | Quantity
10 | 2
20 | 2
10 | 6
21 | 1

Then sorting by cheapest price and most quantity would give:
10 | 6
10 | 2
20 | 2
21 | 1

Reply:
In this case, it should have been just in random order instead of making it increasingly more difficult to transfer it through and http request, consider it is a params, not a json data

Reply:
`/shop?sort_by=price&sort_by=quantity` solves the issue.

Having it in random works, but if there are 10s of items, having to page through it all to find the one I want is tedious.

Reply:
It wouldn't work on the request side because aiohttp take in a dictionary for params which repeated key is not allowed

Reply:
Ok, then sending a single value will be fine.

Reply:
Thx :)